### PR TITLE
chore: add upstream docx-editor changes + few additional docx fidelity slips

### PR DESCRIPTION
**Pull request by jan-kubica**
Syncs applicable upstream docx-editor fixes into Folio, including DOCX style cascade handling, body-scoped PM position lookups, footnote table rendering, run-level OOXML mark propagation, and layout fidelity fixes.

### CODE REVIEW

File: packages/folio/src/core/layout-bridge/toFlowBlocks.ts [hunk 1]
```diff
@@ -1098,11 +1198,26 @@ function convertTableCell(
...
     | undefined;
+  const resolvePaddingSide = (
+    cellTwips: number | undefined,
+    tableTwips: number | undefined,
+  ): number => {
+    if (cellTwips !== undefined) {
+      const px = twipsToPixels(cellTwips);
+      if (px > 0) {
+        return px;
+      }
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/folio/src/core/layout-bridge/toFlowBlocks.ts:1349]: CC on behalf of @jan-kubica

Keeping this as-is: explicit zero `w:tcMar` falls through to the table/default margin intentionally, matching the upstream docx-editor behavior for Word-compatible table margins. This follow-up adds a regression test for that fallthrough and separately restores the built-in TableNormal left/right fallback when no cell/table margin exists.

File: packages/folio/src/core/layout-bridge/toFlowBlocks.ts [hunk 2]
```diff
@@ -1098,11 +1205,31 @@ function convertTableCell(
...
+    side: TablePaddingSide,
+    cellTwips: number | undefined,
+    tableTwips: number | undefined,
+  ): number => {
+    if (cellTwips !== undefined) {
+      const px = twipsToPixels(cellTwips);
+      if (px > 0) {
+        return px;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/folio/src/core/layout-bridge/toFlowBlocks.ts:1350]: CC on behalf of @jan-kubica

Keeping this behavior as-is: explicit zero `w:tcMar` intentionally falls through to table/default margins for Word-compatible table margin resolution, matching the upstream docx-editor change this PR ports. The PR already has a regression for that fallthrough, and separately restores the built-in TableNormal side padding when no margin source exists.

File: packages/folio/src/core/layout-painter/renderParagraph.ts [hunk 3]
```diff
@@ -139,6 +139,66 @@ function applyRunStyles(element: HTMLElement, run: TextRun | TabRun): void {
     element.style.letterSpacing = `${run.letterSpacing}px`;
   }
 
+  if (run.allCaps) {
+    element.style.textTransform = "uppercase";
+  }
+  if (run.smallCaps) {
+    element.style.fontVariant = "small-caps";
+  }
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/folio/src/core/layout-painter/renderParagraph.ts:148]: CC on behalf of @jan-kubica

Fixed in `665583a2f`. This was real: `smallCaps` was rendered with `font-variant: small-caps`, but measurement, click/selection geometry, cache identity, and render-time tab position tracking did not all carry that variant. I added regressions for small-caps line wrapping, per-character widths, and paragraph cache keys, then threaded the canvas `small-caps` font variant through those measurement paths.

File: packages/folio/src/core/layout-painter/renderParagraph.ts [hunk 4]
```diff
@@ -139,6 +139,66 @@ function applyRunStyles(element: HTMLElement, run: TextRun | TabRun): void {
...
+  if (run.positionPx) {
+    element.style.verticalAlign = `${run.positionPx}px`;
+  }
+  if (run.horizontalScale && run.horizontalScale !== 100) {
+    element.style.display = "inline-block";
+    element.style.transform = `scaleX(${run.horizontalScale / 100})`;
+    element.style.transformOrigin = "left center";
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/folio/src/core/layout-painter/renderParagraph.ts:155]: CC on behalf of @jan-kubica

Fixed in `12fbf3edf`. This was real: after making `w:scale` measurement-aware, the browser still laid out the transformed span at its unscaled inline width. The renderer now reserves the scaled advance on horizontally scaled text and field runs, with a regression covering the scaled text DOM width.

**Comment by jan-kubica** [comment]
Addressed the two latest bot comments:

- `PRRT_kwDOSTHQ0c6AqehC`: fixed in f4039fe3997365b20beddbe65dca0e1fffcd1cb8. Real issue: render-time tab tracking measured preceding text without `letterSpacing`, even though DOM paint and layout measurement include it. Added a painter regression for a tab after spaced text. Eigenpal missed this render-path half.
- `PRRT_kwDOSTHQ0c6AqyJt`: fixed in 5fe367eca. Real issue: collapsed trailing `spaceAfter` from the previous block was carried to the top of a new page after pagination advanced. Added regressions proving trailing spacing is dropped across the break while explicit `spaceBefore` on the next block is preserved. Eigenpal missed this pagination edge case too.

Validation: `bun --filter @stll/folio test` (515 pass), `bun --filter @stll/folio lint`, `bun --filter @stll/folio typecheck`; pre-push also passed i18n, format check, typecheck, and lint.

CC on behalf of @jan-kubica

**Comment by jan-kubica** [comment]
Addressed `PRRT_kwDOSTHQ0c6Aq6X-` in 6f2024cce. This was real: style-less tables already fell back to the default table style for `tblPr` borders/margins, but conditional `tblStylePr` lookup still used only the explicit table style id, so default whole-table/banding/cell/run conditionals were skipped. The converter now uses the resolved explicit table style when present, otherwise the default table style, for conditional style resolution. Added a regression covering a style-less table receiving default whole-table shading and bold run formatting. Eigenpal missed this default-style conditional path.

Validation: `bun --filter @stll/folio test` (516 pass), `bun --filter @stll/folio lint`, `bun --filter @stll/folio typecheck`; pre-push also passed i18n, format check, typecheck, and lint.

CC on behalf of @jan-kubica

**Comment by jan-kubica** [comment]
Addressed `PRRT_kwDOSTHQ0c6Aq-ur` in de862f489. This was real: after rendering DOCX bold as numeric `font-weight: 800`, layout and render-time canvas measurement still used CSS `bold`/700. I moved the calibrated weight into a shared constant and now both `buildFontString` and the render-time tab measurer use the same weight as DOM painting. Added regressions for the layout font string and for tab tracking after bold text. Eigenpal missed this measurement/render parity detail.

Validation: `bun --filter @stll/folio test` (518 pass), `bun --filter @stll/folio lint`, `bun --filter @stll/folio typecheck`; pre-push also passed i18n, format check, typecheck, and lint.

CC on behalf of @jan-kubica

**Comment by jan-kubica** [comment]
Addressed two unresolved bot threads:

- `PRRT_kwDOSTHQ0c6ArGJR` in 5fb18e56b. This was real: structured footnote content was rendered with a dummy `RenderContext`, so `PAGE` / `NUMPAGES` fields in footnotes became `0`. `renderFootnoteArea` now receives the real page context, with a regression for `Page 7 of 9`. Eigenpal missed this structured-footnote field context path.
- `PRRT_kwDOSTHQ0c6ArK5k` in 42b666079. This was also real: an explicit DOCX run toggle like `<w:b w:val="0"/>` was converted to absence of a mark, which layout then interpreted as “inherit paragraph default.” I added an explicit run-formatting override mark for negative/default-off values, thread it through PM conversion, layout, editor mark utilities, and DOCX export, with a regression covering bold/italic/allCaps under paragraph style defaults. Eigenpal missed the inherit-vs-explicit-off distinction.

Validation: `bun test packages/folio/src/core/layout-bridge/toFlowBlocks-style-cascade.test.ts -t "explicit run formatting toggles"`, `bun test packages/folio/src/core/layout-painter/renderPage.test.ts`, `bun --filter @stll/folio test` (520 pass), `bun --filter @stll/folio lint`, `bun --filter @stll/folio typecheck`; pre-push also passed i18n, format check, typecheck, and lint.

CC on behalf of @jan-kubica

**Comment by jan-kubica** [comment]
Addressed `PRRT_kwDOSTHQ0c6ArbxT` in 89852ace7. This was real: an explicit table style that existed but omitted `tblPr.borders` / `tblPr.cellMargins` could still fall back to the document default table style, so a borderless explicit style would inherit default-grid borders and padding. The default table style is now used only for unstyled tables; explicit styles use their own table properties without that fallback. Added a regression covering an explicit `Borderless` style in a document whose default table style has grid borders and margins. Eigenpal missed this explicit-style-vs-unstyled-table distinction.

Validation: `bun test packages/folio/src/core/layout-bridge/toFlowBlocks-style-cascade.test.ts -t "explicit table styles"`, full `toFlowBlocks-style-cascade.test.ts`, `bun --filter @stll/folio test` (521 pass), `bun --filter @stll/folio lint`, `bun --filter @stll/folio typecheck`; pre-push also passed i18n, format check, typecheck, and lint.

CC on behalf of @jan-kubica
